Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,4 +1,44 @@
-function setAlarm() {}
+let countdown
+let timeLeft
+
+function setAlarm() {
+  const timeRemaining = document.getElementById("timeRemaining");
+  const alarmInput = document.getElementById("alarmSet");
+  timeLeft = parseInt(alarmInput.value);
+
+  if (isNaN(timeLeft) || timeLeft <= 0) {
+    alert("Please enter a valid number which must be greater than 0");
+  }
+
+  if (countdown) {
+    clearInterval(countdown);
+  }
+
+  function updateDisplay() {
+    const minutes = Math.floor(timeLeft / 60);
+    const seconds = timeLeft % 60;
+    timeRemaining.textContent = `Time Remaining: ${String(minutes).padStart(
+      2,
+      "0"
+    )}:${String(seconds).padStart(2, "0")}`;
+  }
+
```

Comment:
Looks great, Does your clock working in browser? I checked it by adjusting some code (not your code though).